frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/RediscoverSuggester.java

```diff
@@ -0,0 +1,91 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.table;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils.getFileModificationDate;
+import static com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils.setFileModificationDate;
+
+import java.nio.file.Path;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.MessageDialogBuilder;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.ui.GuiUtils;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import lombok.CustomLog;
+import lombok.RequiredArgsConstructor;
+
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+
+@CustomLog
+@RequiredArgsConstructor
+public class RediscoverSuggester {
+
+  private final Project project;
+
+  private final GitRepository gitRepository;
+
+  private final Runnable discoverOperation;
+
+  // TODO (#270): a candidate for custom settings tab
+  private final int DAYS_AFTER_WHICH_TO_SUGGEST_DISCOVER = 14;
+
+  public void performIfNotDeclined() {
+    var macheteFilePath = Option.of(gitRepository).map(GitVfsUtils::getMacheteFilePath).getOrNull();
+    if (macheteFilePath == null) {
+      LOG.warn("Cannot proceed with rediscover suggestion workflow - selected machete file is null");
+      return;
+    }
+
+    var lastModifiedTimeMillis = getFileModificationDate(macheteFilePath).getOrNull();
+    if (lastModifiedTimeMillis == null) {
+      LOG.warn("Cannot proceed with rediscover suggestion workflow - could not get file modification date");
```

Comment:
Seems that `string.GitMachete.RediscoverSuggester.failed-to-get-file-attrs` is unused... not sure about other strings marked as unused by IntelliJ (maybe false negatives? as for the `*.GitMacheteToolbar.*` ones):

![image](https://user-images.githubusercontent.com/3383210/98406537-749efd00-206e-11eb-8594-ae07cbfc3d95.png)